src/main/java/de/htwberlin/GradesOverview.java

```diff
@@ -14,14 +18,90 @@ public GradesOverview() {
 
   public void add(int grade) {
     // TODO
+  try{
+    if(!((grade == 1) || (grade == 2) || (grade== 3) || (grade== 4) || (grade== 5))) {
+      throw new IllegalArgumentException("Dies ist keine gültige Note");
+    }
+    grades[index] = grade;
+    for(int i=0; i<=index; i++){
+      System.out.println(grades[i]);
+    }
+    index++;
   }
 
+      catch (ArrayIndexOutOfBoundsException e){
+      System.out.println("Die max Anzahl Noten wurde überschritten");
+      }
+  calcAverage();
```

Comment:
Die Idee wäre hier, dass das Array einfach vergrössert wird. 

Sobald 10 Elemente im Array enthalten sind, erstellt man einfach ein neues und vor allem grösseres Array und kopiert die Elemente aus dem kleineren Array in das neue grössere Array:

```java
  public void add(int grade) {
    validateGrade(grade);
    int index = lastIndex + 1;
    if (index >= grades.length) {
      resize();
    }
    grades[index] = grade;
    lastIndex = index;
  }

  private void validateGrade(int grade) {
    if (grade <= 0 || grade > 5) {
      throw new IllegalArgumentException("Only grades between 1 and 5 are allowed. Input was: " + grade);
    }
  }

  private void resize() {
    grades = Arrays.copyOf(grades, grades.length + 5);
  }
```